app/controllers/paypal_accounts_controller.rb

```diff
@@ -71,7 +73,13 @@ def new
     })
   end
 
-  def create
+  def ask_order_permission
+    return redirect_to action: :new unless PaypalHelper.community_ready_for_payments?(@current_community)
+
+    create_paypal_account
```

Comment:
This might not be entirely about new code in this branch but I'm having hard time understanding the need to have separate create_paypal_account method as it stands. This is the only place it is called from and this action just transforms control of flow to a private method. Make sit hard to understand what are the possible outcomes of invoking this action. A better style would be to move the code that decides what to flash and render into this action and have a helper for fetching redirect_url from PayPal.